src/excmds.ts

```diff
@@ -328,6 +328,18 @@ export async function updatenative(interactive = true) {
     }
 }
 
+/**
+ *  Restarts firefox with the same commandline arguments
+ *  Warning: This can kill your tabs, especially if you :restart several times
+ *  in a row
+ */
+//#background
+export async function restart() {
+    const firefox = (await Native.ffargs()).join(" ")
+    const ppid = (await Native.pyeval("os.getppid()")).content
+    console.log(await Native.run(`kill -15 ${ppid} && ${firefox}`))
```

Comment:
Please use `logger.log` : )

Reply:
Hi Everyone,

```Javascript
await Native.run(`kill -15 ${ppid} && ${firefox}`)
```
As compared to above, something like the following might be more cross-platform friendly:

```Javascript
await Native.pyeval("os.kill(os.getppid(), signal.SIGNAL_SIGTERM)")
await Native.run(`${firefox}`)
```

Thanks.

Reply:
That second line might not get a chance to run (it might not get a chance in the original either, if firefox kills its children or the shell listens for its parent's death).

Other point noted :)

Reply:
This is the old version, the current version does things a bit differently (see https://github.com/cmcaine/tridactyl/pull/510/files ), but your concerns still stand.

I doubt restart can be properly supported on windows for now as there's currently no way to access the commandline arguments Firefox was started with and thus no way to restart it with the proper arguments. Getting the command line arguments is [possible on windows](https://stackoverflow.com/questions/17563411/how-to-get-command-line-info-for-a-process-in-powershell-or-c-sharp) but requires shelling out to powershell, I can't implement this for now as I don't have access to a windows computer.

There's also the problem of waiting for the lock file to be removed. If done in python, this would probably require a lot more code, and thus maybe deserve to be directly implemented in the native messenger.